drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -503,7 +507,7 @@ local function do_configure(driver, device)
     end
     profile_name = profile_name .. create_air_quality_sensor_profile(device)
 
-  elseif #thermostat_eps > 0 then
+  elseif #thermostat_eps > 0 or device_type == TMST_DEVICE_TYPE_ID then
```

Comment:
the get_device_type logic currently defaults to thermostat when no other device type (i.e. fan, air purifier, room ac) are found. Including this logic could jeopardize that flow.

Reply:
I updated my pull request by incorporating thermostat check logic into the "device_type == FAN_DEVICE_TYPE_ID" condition to handle situations where a device supports both fan and thermostat types.